### PR TITLE
Shadhrush - Fixing "Send Emails" visibility for Volunteers with 'Other Links' access and adding a new Send Emails Permission.

### PR DESCRIPTION
**Pull request by Shadhrush5**
# Description
Screenshot 2024-08-15 at 12 35 01 PM
Screenshot 2024-08-15 at 12 41 17 PM

## Related PRS (if any):
This frontend PR is related to the backend PR https://github.com/OneCommunityGlobal/HGNRest/pull/1082.
…

## Main changes explained:
1. Added a new permission for access to 'Send Mails'.
2. Added a new route called 'Announcements' and 'Send Mails' permission comes under this.
3. Gave 'Owner' the new permission by default. For the rest it can be added.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin/owner user
5. Give the 'add Roles' permission to any Volunteer.
6. log on as that volunteer who has the permission.
7. go to dashboard→ Other links
8. You will not see 'Send Mails' anymore
9. log as admin/owner user
5. Give the 'Send Emails' permission to the Volunteer.
6. log on as that volunteer who has the permission.
7. go to dashboard→ Other links
8. You will now see 'Send Mails' and click on the same to verify the right page opens.

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/8e03c11b-0f10-4974-b45f-e9bfd240ff01

https://github.com/user-attachments/assets/feb97799-7994-4a4a-bf3c-2042fa407b38

### CODE REVIEW

**Review by Pranay-ai — DISMISSED** [review]
Hey @Shadhrush5 
I have tested the PR by following the steps and its working as expected 
image

Thank You

**Review by nikhilpittala16 — APPROVED** [review]
I have tested the PR and everything works as expected.
Screenshot (169)
Screenshot (170)

**Comment by Lemonnycodes** [comment]
Hi,

The PR Works as described. 
 Both the "Send Mails" and "Add Roles" permissions are functioning correctly. When these permissions are assigned to a volunteer, the corresponding options appear under "Other Links" on the dashboard. The permissions are applied as expected, and the UI behaves as intended. No issues were encountered during testing.
 
Screenshots:
PR (1) 2572
PR (2) 2572

**Review by vishavk1992 — DISMISSED** [review]
I have tested this PR and i gave permission to my volunteer account to send email. and it successfully display on my volunteer account. 

Screenshot 2024-08-16 165332
Screenshot 2024-08-16 165348

**Review by AnkitLall — APPROVED** [review]
I have tested the feature, following the instructions in the PR, and it works as expected. I’ve attached a video recording of the testing process.


https://github.com/user-attachments/assets/1e99cd8b-59e0-4253-98f2-8f2d6da5d39c

**Review by JatinAgrawal94 — DISMISSED** [review]
Everything works as expected.

https://github.com/user-attachments/assets/62b038ba-9094-4e9f-9673-6197c5a24395

**Review by Aishwarya0898 — APPROVED** [review]
Hi @Shadhrush5 
The PR is working as expected. Here is the screenshot for reference.


https://github.com/user-attachments/assets/68b39c84-a829-47cc-8963-66afcb312bec

**Review by SMitta11 — APPROVED** [review]
I login as admin first and gave volunteer user permission to send emails and then login as volunteer and can see the send email.
It is working as expected.
Screenshots:
Screen Shot 2024-08-17 at 1 42 57 AM

Screen Shot 2024-08-17 at 1 44 58 AM

**Review by adityasure9 — APPROVED** [review]
I have tested the PR following the provided steps, and it works as expected. The 'Send Mails' permission is correctly applied, and the UI behaves as intended when permissions are assigned to a volunteer. Everything functions as described. Great work!

image

**Review by nathanah — CHANGES_REQUESTED** [review]
Everthing looks good here, but this PR needs a backend equivalent to reject API requests when users don't have the relevant permission. The relevant functions seem to be `sendEmail` and `sendEmailToAll` in `emailController.js`. Also please add this permission to the [permissions spreadsheet](https://docs.google.com/spreadsheets/d/1TfiJY9OLDZuyP2UgF0C1wI9tOSLKbS9MiJFH6n1Dsao/edit?gid=0#gid=0).

**Review by Gauravmg6 — APPROVED** [review]
Everything works well!

Screenshot 2024-08-18 002559

**Review by Parth-tech — APPROVED** [review]
Hi Shadrush,

The PR works correctly. Once the volunteer is assigned the sendEmail permission, user is able to see that option and navigate to the page too.

`Image References:`


Screenshot 2024-08-21 at 7 21 48 PM
Screenshot 2024-08-21 at 7 19 50 PM
Screenshot 2024-08-21 at 7 21 55 PM
Screenshot 2024-08-21 at 7 22 19 PM
Screenshot 2024-08-21 at 7 23 00 PM

**Review by skaush21 — DISMISSED** [review]
Retested the 'Send Emails' function, everything works as intended. Implemented code looks good. Tested using backend [PR 1082](https://github.com/OneCommunityGlobal/HGNRest/pull/1082/files)

**Review by vishavk1992 — COMMENTED** [review]
I approved this PR .

**Review by nikhilpittala16 — DISMISSED** [review]
I have retested the PR and everything works as expected.

**Review by SMitta11 — APPROVED** [review]
Initially volunteer account could not see send email and when I gave permission of 'send email' to volunteer and log on as that volunteer who has the permission then send email was visible and the correct page opens up on clicking.
Screenshots:
Screen Shot 2024-08-23 at 12 13 52 AM
Screen Shot 2024-08-23 at 12 07 29 AM
Screen Shot 2024-08-23 at 12 15 36 AM
Screen Shot 2024-08-23 at 12 15 46 AM

**Review by AJAYINAVOLU — COMMENTED** [review]
Changes are working as expected.




Thank You

**Review by manikittu810 — DISMISSED** [review]
Hello @Shadhrush5 
I have followed the mentioned steps. Everything is working as mentioned.

https://github.com/user-attachments/assets/c5ccc0fd-ad0c-4ecd-8016-0ecd17e7d5ca

Thank you.

**Review by sri0606 — APPROVED** [review]
Hello @Shadhrush5 , 
I have reviewed the changes. 

Test notes:
* At first, **Other Links** option was not visible on the volunteer account.
* Then, I gave the same account "Send Emails" permission.
* Now, the **Other Links** option is visible and the dropdown menu has "Send emails" option and onClick, navigates to correct page.

Screenshot 2024-08-30 123523
Screenshot 2024-08-30 123803

**Comment by Shadhrush5** [comment]
> Everthing looks good here, but this PR needs a backend equivalent to reject API requests when users don't have the relevant permission. The relevant functions seem to be `sendEmail` and `sendEmailToAll` in `emailController.js`. Also please add this permission to the [permissions spreadsheet](https://docs.google.com/spreadsheets/d/1TfiJY9OLDZuyP2UgF0C1wI9tOSLKbS9MiJFH6n1Dsao/edit?gid=0#gid=0).

Changes addressed and the backend PR is approved by the commenter as well.

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -134,6 +134,8 @@ export function Header(props) {
   const canAccessPopups =
     props.hasPermission('createPopup', !isAuthUser && canInteractWithViewingUser) ||
     props.hasPermission('updatePopup', !isAuthUser && canInteractWithViewingUser);
+  // SendEmails
+  const canAccessSendEmails = props.hasPermission('sendEmails', !isAuthUser);
```
Comment (nathanah) [inline src/components/Header/Header.jsx:138]:
```suggestion
  const canAccessSendEmails = props.hasPermission('sendEmails');
```
hasPermission only accepts one argument, right?

**Review by nathanah — APPROVED** [review]

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**